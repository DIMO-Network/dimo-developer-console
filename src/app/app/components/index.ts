export * from './OnboardingCard';
export * from './Onboarding';
