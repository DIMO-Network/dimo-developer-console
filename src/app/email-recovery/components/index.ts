export * from './EmailRecoveryForm';
export * from './CheckEmail';