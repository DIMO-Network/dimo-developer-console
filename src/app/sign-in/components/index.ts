export * from './OtpInputForm';
export * from './PasskeyLogin';
export * from './SignInMethodForm';
