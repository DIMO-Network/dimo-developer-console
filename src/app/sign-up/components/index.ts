export * from './BuildForForm';
export * from './CompanyInfoForm';
export * from './SignUpWith';
export * from './UserInfoForm';
