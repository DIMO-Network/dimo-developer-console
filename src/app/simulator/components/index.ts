export { View } from './View';
export { Header } from './Header';
export { SimulatedVehiclesSection } from './SimulatedVehiclesSection';
export { CreateSimulatedVehicleModal } from './CreateSimulatedVehicleModal';
