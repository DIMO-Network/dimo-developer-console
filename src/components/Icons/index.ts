export * from './GoogleIcon';
export * from './GitHubIcon';
