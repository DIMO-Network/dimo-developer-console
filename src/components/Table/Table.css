.table {
  @apply min-w-full;

  .table-body {
    @apply divide-y divide-dark-grey-950;
  }

  .table-action-column {
    @apply relative py-3.5 pl-3 pr-4 sm:pr-0;
  }

  .table-action-cell {
    @apply relative whitespace-nowrap px-3 py-1 text-right text-xs inline-flex gap-4 justify-center items-center;
  }
}
