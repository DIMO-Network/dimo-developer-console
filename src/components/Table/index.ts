export * from './Table';
export { PaginatedTable } from '@/components/Table/PaginatedTable';
