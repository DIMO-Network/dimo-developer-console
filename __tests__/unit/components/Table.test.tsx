import { render, screen } from '@testing-library/react';

import { Table } from '@/components/Table';
import { appListMock } from '@/mocks/appList';
import { ContentCopyIcon } from '@/components/Icons';

describe('Table', () => {
  it('renders a table with basic config', () => {
    render(
      <Table
        columns={[
          {
            name: 'name',
          },
        ]}
        data={appListMock.map(({ name }) => ({ name }))}
      />,
    );

    const titleElm = screen.getByText('name');
    const appNameElm = screen.getByText(appListMock[0].name);

    expect(appNameElm).toBeInTheDocument();
    expect(titleElm).toBeInTheDocument();
  });

  it('renders a table with custom render', () => {
    const { container } = render(
      <Table
        columns={[
          {
            name: 'name',
            render: (item: Record<string, string>) => {
              return (
                <p>
                  {item.name ?? ''} - {item.scope ?? ''}
                </p>
              );
            },
          },
          { name: 'names' },
        ]}
        data={appListMock.map(({ name, scope }) => ({ name, scope }))}
      />,
    );

    const appNameElm = screen.getByText(
      `${appListMock[0].name} - ${appListMock[0].scope}`,
    );

    expect(appNameElm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders a table with actions', () => {
    const renderAction = () => (
      <ContentCopyIcon className="2-5 h-5 fill-white" />
    );
    const { container } = render(
      <Table
        columns={[
          {
            name: 'name',
          },
        ]}
        actions={[renderAction]}
        data={appListMock.map(({ name, scope }) => ({ name, scope }))}
      />,
    );

    const actionElms = screen.getAllByRole('content-copy-icon');

    expect(actionElms).toHaveLength(appListMock.length);
    expect(container).toMatchSnapshot();
  });
});
