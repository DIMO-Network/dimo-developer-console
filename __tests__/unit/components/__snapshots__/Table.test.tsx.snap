// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders a table with actions 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          class="custom-table-column"
          scope="col"
        >
          name
        </th>
        <th
          class="table-action-column"
          scope="col"
        >
          <span
            class="sr-only"
          >
            Actions
          </span>
        </th>
      </tr>
    </thead>
    <tbody
      class="table-body"
    >
      <tr>
        <td
          class="table-cell"
        >
          Test App
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          Fleet Management App
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          Data Analytics
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          Hello World
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          Hello World
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          Hello World
        </td>
        <td
          class="table-action-cell"
        >
          <svg
            class="2-5 h-5 fill-white"
            fill="none"
            role="content-copy-icon"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.7497 0.833374H3.74967C2.83301 0.833374 2.08301 1.58337 2.08301 2.50004V14.1667H3.74967V2.50004H13.7497V0.833374ZM16.2497 4.16671H7.08301C6.16634 4.16671 5.41634 4.91671 5.41634 5.83337V17.5C5.41634 18.4167 6.16634 19.1667 7.08301 19.1667H16.2497C17.1663 19.1667 17.9163 18.4167 17.9163 17.5V5.83337C17.9163 4.91671 17.1663 4.16671 16.2497 4.16671ZM16.2497 17.5H7.08301V5.83337H16.2497V17.5Z"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table renders a table with custom render 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          class="custom-table-column"
          scope="col"
        >
          name
        </th>
        <th
          class="custom-table-column"
          scope="col"
        >
          names
        </th>
      </tr>
    </thead>
    <tbody
      class="table-body"
    >
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Test App
             - 
            sandbox
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Fleet Management App
             - 
            production
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Data Analytics
             - 
            production
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Hello World
             - 
            production
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Hello World
             - 
            production
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
      <tr>
        <td
          class="table-cell"
        >
          <p>
            Hello World
             - 
            production
          </p>
        </td>
        <td
          class="table-cell"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
